x/mint/keeper/hooks.go

```diff
@@ -21,20 +21,22 @@ func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumb
 		if epochNumber < params.MintingRewardsDistributionStartEpoch {
 			return
 		} else if epochNumber == params.MintingRewardsDistributionStartEpoch {
-			k.SetLastHalvenEpochNum(ctx, epochNumber)
+			k.SetLastReductionEpochNum(ctx, epochNumber)
 		}
 		// fetch stored minter & params
 		minter := k.GetMinter(ctx)
 		params := k.GetParams(ctx)
 
 		// Check if we have hit an epoch where we update the inflation parameter.
-		// Since epochs only update based on BFT time data, it is safe to store the "halvening period time"
-		// in terms of the number of epochs that have transpired.
-		if epochNumber >= k.GetParams(ctx).ReductionPeriodInEpochs+k.GetLastHalvenEpochNum(ctx) {
-			// Halven the reward per halven period
+		// We measure time between reductions in number of epochs.
```

Comment:
Question: How do we determine an epoch has passed? Is it based on specific block height or time passed? 

Does epoch passing has any correlation with how fast blocks are produced?

Reply:
I think the answers to these are documented here:

https://github.com/osmosis-labs/osmosis/blob/main/x/epochs/spec/README.md#L24-L35

Please let me know if anything goes unanswered